src/checkin/services/flow/step/GetExtraPassengerInformationStep.spec.ts

```diff
@@ -0,0 +1,108 @@
+    })
+  })
+
+  describe('Test at execution', () => {
+    it('should request the extra passengers information', async () => {
+      const context: Context = wireContextMockWithNumberOfPassengersInSessionData(2)
+      wireContextMockWithNullExtraPassengerInformationInRequest(context)
+
+      const response = await step.onExecute(context)
+
+      expect(response).toBeFalsy()
+      expect(context.getResponse().requiredFiles?.extra_passenger_information).toBeNull()
+      expect(context.getSession().extra_passenger_information).toBeUndefined()
+    })
+
+    it('should request the exact amount of extra passengers information', async () => {
+      const context = wireContextMockWithNumberOfPassengersInSessionData(2)
+      wireContextMockWithAmountOfExtraPassengersInformationInRequest(0, context)
+
+      const response = await step.onExecute(context)
+
+      expect(response).toBeFalsy()
+      expect(context.getResponse().requiredFiles?.extra_passenger_information).toBeNull()
+      expect(context.getSession().extra_passenger_information).toBeUndefined()
+    })
+
+    it('should set the extra passenger information in session if it is given by the user', async () => {
+      const context = wireContextMockWithNumberOfPassengersInSessionData(2)
+      wireContextMockWithAmountOfExtraPassengersInformationInRequest(1, context)
+      const mockExtraPassengerInformation = context.getRequest().fields?.extra_passenger_information as Array<{ name: string, passport: string }>
+
+      const response = await step.onExecute(context)
+
+      expect(response).toBeTruthy()
+      expect(context.getSession().extra_passenger_information).toBe(mockExtraPassengerInformation)
+    })
+  })
+})
+
+function wireContextMockWithNumberOfPassengersInSessionData(numberOfPassengers: number, context = new MockContext()): Context {
+  context.withSessionBuilder(sessionBuilder => sessionBuilder
+    .data({
+      country: context.getSession().data.country,
+      flights: context.getSession().data.flights,
+      passengers: numberOfPassengers,
+      agreementSigned: context.getSession().data.agreementSigned
+    })
+  )
+  return context
+}
+
+function wireContextMockWithNullExtraPassengerInformationInRequest(context = new MockContext()): Context {
+  context.withRequestBuilder(requestBuilder => requestBuilder.fields({ extra_passenger_information: null }))
+
+  return context
+}
+
+function wireContextMockWithAmountOfExtraPassengersInformationInRequest(amountOfExtraPassengersInformation: number, context = new MockContext()): Context {
+  const mockExtraPassengerInformation = buildMockExtraPassengerInformation(amountOfExtraPassengersInformation)
+
+  context.withRequestBuilder(requestBuilder => requestBuilder.fields({ extra_passenger_information: mockExtraPassengerInformation }))
+
+  return context
+}
+
+function buildMockExtraPassengerInformation(amountOfExtraPassengersInformation: number): Array<{ name: string, passport: string }> {
```

Comment:
The type { name: string, passpass: string } should be encapsulated (in an interface or typescript type) to avoid inconsistencies elsewhere

Reply:
Totally right, thanks.